executor/runner/runner.go

```diff
@@ -470,3 +474,61 @@ func (r *Runner) getPlatformSidecarStatuses() []corev1.ContainerStatus {
 	}
 	return statuses
 }
+
+func (r *Runner) monitorKernelLogsForStuckTasks(ctx context.Context, task Task) {
+	l := logger.G(ctx)
+	ch := make(chan string, 20)
+
+	go func() {
+		kmsg, err := os.Open("/dev/kmsg")
+		if err != nil {
+			l.Info("couldn't open /dev/kmsg (%v), not monitoring for stuck tasks", err)
+			close(ch)
+			return
+		}
+		defer kmsg.Close()
+
+		// seek to the end of the file... otherwise kmsg will happily
+		// tell us about everything in its buffer, which we do not
+		// really care about.
+		_, err = kmsg.Seek(0, io.SeekEnd)
+		if err != nil {
+			close(ch)
+			l.Info("couldn't seek /dev/kmsg (%v), not monitoring for stuck tasks", err)
+			return
+		}
+
+		scanner := bufio.NewScanner(kmsg)
+		for scanner.Scan() {
+			line := scanner.Text()
+			ch <- line
+		}
+
+		if scanner.Err() != nil {
+			l.Info("problem reading from /dev/kmsg: %v", err)
+			close(ch)
+		}
+	}()
+
+	go func() {
+		for {
+			select {
+			case <-r.killChan:
+				return
+			case kernelLogLine, more := <-ch:
+				msg, err := handleKernelLogLine(kernelLogLine, task)
```

Comment:
I think this might read better like

if isStuckTaskLogLine(kernelLogLine) {
  parseStuckKernelLogLine(kernelLogLine)
}
And can give us more room in the future for other types of lines, and makes it more clear what that function is doing exactly. And it doesn't have to worry about so many error cases as much